### PR TITLE
Remove a few unnecessary CCI interfaces

### PR DESCRIPTION
**Pull request by tmat**
IMetadataConstant, IMetadataTypeOf, IMetadataCreateArray only have a single implementation.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler

**Comment by AlekseyTs** [comment]
@tmat Compiler changes require two sign-offs from the compiler team.

File: src/Compilers/Core/Portable/PEWriter/Members.cs [hunk 1]
```diff
@@ -718,7 +718,7 @@ internal interface IPropertyDefinition : ISignature, ITypeDefinitionMember
         /// <summary>
         /// A compile time constant value that provides the default value for the property. (Who uses this and why?)
         /// </summary>
-        IMetadataConstant DefaultValue
+        MetadataConstant DefaultValue
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/PEWriter/Members.cs]: >DefaultValue [](start = 25, length = 12)

Is this property used at all?

Comment (tmat) [inline src/Compilers/Core/Portable/PEWriter/Members.cs]: I don't know. Will check.

Comment (tmat) [inline src/Compilers/Core/Portable/PEWriter/Members.cs]: No, will delete it.

**Pull request merged**